source/npm/check-name.js

```diff
@@ -0,0 +1,13 @@
+'use strict';
+const npmName = require('npm-name');
+
+module.exports = async pkg => {
```

Comment:
it seems a little confusing to me. 
could this be `module.exports.checkName = ...`, so the usage of the functions would be more consistent:
```js
const {isExternalRegistry} = require('./npm/check-name');
const {checkName} = require('./npm/check-name');
```

Reply:
Also, on second thought, adding a new file just for the name check seems a bit unnecessary. I think I'll apply your suggestion as well as move the functions in `source/npm/check-name.js` to `source/npm-utils.js`.